openforcefield/utils/callback.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python
+
+"""
+Utility classes and functions to create objects supporting callback registration.
+
+"""
+
+__all__ = [
+    'callback_method',
+    'CallbackRegistrationError',
+    'Callbackable',
+]
+
+
+# =====================================================================
+# GLOBAL IMPORTS
+# =====================================================================
+
+import functools
+import inspect
+
+
+# =====================================================================
+# CALLBACKABLE CLASSES
+# =====================================================================
+
+def callback_method(func=None, events=()):
+    """Decorator used to mark a method as callbackable.
+
+    This decorator is designed to be used by ``Callbackable`` children
+    classes. Usage examples can be found in the documentation of ``Callbackable``.
+
+    Parameters
+    ----------
+    func : MethodType
+        The instance method to decorate. Static and class methods are
+        currently not supported.
+    events : Iterable[str]
+        The name of the callback events associated to this method.
+
+    See Also
+    --------
+    Callbackable
+
+    """
+
+    if func is None and len(events) > 0:
+        # Decorator called with @callback_method(events=[...]).
+        return functools.partial(callback_method, events=events)
```

Comment:
I'm not sure that I understand when this is triggered, but I'm glad we have defined behavior for it.

Reply:
This is triggered when the decorator is used as
```python
@callback_method(events=['my_event])
def func():
    ...
```
instead of
```python
@callback_method
def func():
    ...
```